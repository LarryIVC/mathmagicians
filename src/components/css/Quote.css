.cont-loader-msg {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.cont-quote {
  display: flex;
  flex-direction: column;
  padding: 10px;
  margin: 5px;
  max-width: 500px;
  font-size: 0.65em;
  background-color: #eddba9;
  border-radius: 25px;
}

.p-quote {
  font-weight: 400;
  font-style: italic;
  text-align: justify;
}

.p-author {
  text-align: end;
  margin: 5px;
}
