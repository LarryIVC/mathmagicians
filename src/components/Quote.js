import React, { useState, useEffect } from 'react';
import Loader from './Loader';
import './css/Quote.css';

const API_URL = 'https://api.api-ninjas.com/v1/quotes?category=';
const API_KEY = '5Mkq9zqFuG8L7WItA+cbfw==TzJclNtQ7MnlRbxq';

const Quote = () => {
  const [data, setData] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    const category = 'history';
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const response = await fetch(`${API_URL}${category}`, {
          headers: {
            'X-Api-Key': API_KEY,
          },
        });
        if (!response.ok) {
          setHasError(true);
        } else {
          const responseData = await response.json();
          setData(responseData);
        }
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, []);

  if (hasError) return <div data-testid="quote">Something went wrong!</div>;

  if (isLoading) {
    return (
      <div data-testid="quote" className="cont-loader-msg">
        Loading...
        <Loader />
      </div>
    );
  }

  return (
    <article data-testid="quote" className="cont-quote">
      {data.length > 0 ? (
        <>
          <p className="p-quote">{`"${data[0].quote}"`}</p>
          <p className="p-author">
            <strong>Author: </strong>
            {data[0].author}
          </p>
        </>
      ) : (
        <p>No data available</p>
      )}
    </article>
  );
};

export default Quote;
