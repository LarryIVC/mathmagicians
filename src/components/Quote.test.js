import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Quote from './Quote';

describe('Test suite for the Quote component', () => {
  let myQuote;
  beforeEach(async () => {
    render(<Quote />);
    await waitFor(() => {
      myQuote = screen.getByTestId('quote');
    }, { timer: 3000 });
  });

  it('Should test if Quote component is inside the document', () => {
    expect(myQuote).toBeInTheDocument();
  });

  it('Should render a snapshot of the Quote component', () => {
    expect(myQuote).toMatchSnapshot();
  });
});
