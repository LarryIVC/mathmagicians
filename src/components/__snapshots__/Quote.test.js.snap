// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Quote component Should render a snapshot of the Quote component 1`] = `
<div
  class="cont-loader-msg"
  data-testid="quote"
>
  Loading...
  <div
    class="lds-spinner"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
